### ISSUE TITLE
Date instructions for dates without parser

### ISSUE THREAD

**Issue by atomrab** [periodo/periodo-client#89]
Since the ISO standard includes a zero year, and therefore 2 BC is -1, how do we make it clear to users who are entering BC dates in, for example, other languages (e.g. a.C., which doesn't parse), that they need to add positive 1 to all their BC dates?

On the same topic, our BP parsing has ignored the zero year (so 10000 BP parses as -8050). This is fine for real BP dates, which are usually so far in the past that the year doesn't matter. But because Fasti and now ARIADNE are using BP (2000) for all their dates, including more recent ones where a year in either direction will matter, should we insist on the zero year for this particular format? This is  somewhat urgent issue, because we're about to start putting in ARIADNE dates and need to make sure we get this right.

**Comment by rybesh** [comment]
1. Is there anything we can do beyond having some help text that says "ISO year values are permitted to have the value `0`. A year value of `0` represents 1 BCE, `−1` represents 2 BCE, etc."?

2. Currently 1 BCE and 1950 BP both parse to ISO 8601 year `0`. You're saying that the former is correct, but the latter should parse to ISO 8601 year `-1`, and 1951 BP should parse to year `0`? In that case, 1 BP will parse to `1950` and 0 BP will parse to `1951`... is that correct?

**Comment by atomrab** [comment]
>Is there anything we can do beyond having some help text that says "ISO year values are permitted to have the value 0. A year value of 0 represents 1 BCE, −1 represents 2 BCE, etc."?

Nope. I just think we need to put it somewhere where they're going to see it, or we're going to get a lot of inconsistent data when and if we build an international user community.

>Currently 1 BCE and 1950 BP both parse to ISO 8601 year 0. You're saying that the former is correct, but the latter should parse to ISO 8601 year -1, and 1951 BP should parse to year 0? In that case, 1 BP will parse to 1950 and 0 BP will parse to 1951... is that correct?

Not exactly. I'm just trying to figure out how to deal with the ARIADNE BP (2000) dates, all of which I'm going to have to manually convert to ISO dates. I'm not worried about the positive ones, but here's my problem with the BC ones. If I put in "500 BC", the zero year means that my ISO date is -499. If I want to express 500 BC as BP, I add 2000 (if I'm Fasti) or 1950 (if not), and I get 2500 BP(2000), or 2450 BP(1950). If I then put the latter into the parser, I get ISO -500, which is not the same as ISO -499.

With the ARIADNE dates, a lot of them have been converted from original BC dates simply by adding 2000. So do I just subtract 2000 and enter the result as a negative number (e.g. original is 500 BC, +2000 = 2500 BP; 2500-2000 = 500; enter -500)? Or, because my original value was 500 BC, do I subtract 2001? 

I'm happy to treat all the ARIADNE dates in the form 2500 - 2000 = -500, since it's a separate collection; I just don't want to create an unintended inconsistency.

**Comment by rybesh** [comment]
Can we ask Ariadne which of the following they believe?

1. 2500 BP (2000) means the same thing as 50 BC
2. 2500 BP (2000) means the same thing as 51 BC
3. It doesn't matter.

If the answer is 1 or 3, we can treat BP (2000) the same as BP (1950), and we should probably extend the parser to handle any "XXXX BP (YYYY)" unit by subtracting YYYY from XXXX.

If the answer is 2, we should treat "BP (2000)" as an Ariadne-specific thing that really means BP (2001).

What is the most universally understood unit? BCE? Should we consider displaying the BCE-formatted interpretation next to the ISO value so people can verify that they're saying what they mean? This might be more straightforward than writing "ISO year values are permitted to have the value 0. A year value of 0 represents 1 BCE, −1 represents 2 BCE, etc."

**Comment by atomrab** [comment]
>Can we ask Ariadne which of the following they believe?

Done. Will report on the response.

>What is the most universally understood unit? BCE? Should we consider displaying the BCE-formatted interpretation next to the ISO value so people can verify that they're saying what they mean? This might be more straightforward than writing "ISO year values are permitted to have the value 0. A year value of 0 represents 1 BCE, −1 represents 2 BCE, etc."

I think almost all of our contributors -- at least for the immediate future -- will be using the Gregorian calendar, even those writing BP or something else, so yes, BCE is the probably the most recognizable designation (and should be equally intelligible to people using BC). I like the idea of displaying a value in that format during data entry, and I agree that it's probably more effective than a note somewhere, since it will give users the chance to recalibrate on the fly.

**Comment by rybesh** [comment]
Related: #118
